lib/python/rose/config_processors/fileinstall.py

```diff
@@ -253,22 +253,24 @@ def _process(self, conf_tree, nodes, loc_dao, **kwargs):
                 # See if any sources have changed names.
                 if not target.is_out_of_date:
                     conn = loc_dao.get_conn()
-                    try:
-                        prev_dep_locs = conn.execute(
-                            "SELECT * FROM dep_names WHERE name=?",
-                            [target.name]
-                        ).fetchall()
-                        prev_dep_locs = [i[1] for i in prev_dep_locs]
-                        prev_dep_locs = [
-                            loc_dao.select(i) for i in prev_dep_locs
-                        ]
-                        if (
-                            [i.name for i in prev_dep_locs] !=
-                            [i.name for i in target.dep_locs]
-                        ):
-                            target.is_out_of_date = True
-                    finally:
-                        conn.close()
```

Comment:
This is my bad, it turns out that this method can be called multiple times on the same DB (which I should have expected) so closing it here will cause subsequent errors. TBH I'm not sure if/how the DB is closed.